### PR TITLE
Copiers one list 3

### PR DESCRIPTION
**Pull request by arstrube**
Here is "the rest". We make have to break it down....

### CODE REVIEW

File: src/CppUTestExt/MockSupport_c.cpp [hunk 1]
```diff
@@ -72,28 +72,36 @@ static MockExpectedCall* expectedCall = NULL;
 static MockActualCall* actualCall = NULL;
 static MockFailureReporterForInCOnlyCode failureReporterForC;
 
-class MockCFunctionComparatorNode : public MockNamedValueComparator
+class MockCFunctionComparatorAndCopierNode : public MockNamedValueComparator, public MockNamedValueCopier
```
Comment (basvodde) [inline src/CppUTestExt/MockSupport_c.cpp]: Any way we can avoid the use of multiple inheritance here? I think this is the only place in CppUTest and prefer not using it.

File: tests/CppUTestExt/MockSupportTest.cpp [hunk 2]
```diff
@@ -909,6 +939,388 @@ TEST(MockSupportTest, customObjectWithFunctionComparatorThatFailsCoversValueToSt
     CHECK_EXPECTED_MOCK_FAILURE_LOCATION(failure, __FILE__, __LINE__);
 }
 
+TEST(MockSupportTest, customTypeOutputParameterSucceeds)
+{
```
Comment (basvodde) [inline tests/CppUTestExt/MockSupportTest.cpp]: Please remove this kind of comments from the tests. (all of them).

If you want to seperate the AAA sections, just use an empty line. No need to name them.

**Comment by basvodde** [comment]
Mostly good! Lets review some of the comments quickly and get it merged without splitting it in smaller ones.

File: src/CppUTestExt/MockSupport_c.cpp (hunk 1 above)
Comment (arstrube) [inline src/CppUTestExt/MockSupport_c.cpp]: I couldn't think of any. I tried eliminating the extra list, but couldn't
Look at it as these nodes implementing both interfaces. They're nice, clean interfaces and the node implements them. If this was Java, no one would think twice about it.
If you can think of a better solution, I'd be interested. Especially if we didn't need this list at all :)

File: tests/CppUTestExt/MockSupportTest.cpp (hunk 2 above)
Comment (arstrube) [inline tests/CppUTestExt/MockSupportTest.cpp]: No problem. I didn't put them there, I just didn't take the trouble to remove them.

**Comment by arstrube** [comment]
I'll remove the comments tomorrow.

**Comment by arstrube** [comment]
It will be tough about the node:
+ I considered composition, but couldnt think of a good way
+ I considered getting rid of the list, but found no good replacement
+ Ironically, I think two lists would work. I like that less than implementing two interfaces, though.

File: tests/CppUTestExt/MockSupportTest.cpp (hunk 2 above)
Comment (basvodde) [inline tests/CppUTestExt/MockSupportTest.cpp]: Ok. Multiple inheritance is a *potential* problem for old or embedded compilers, so would be good if we can look for a different solution here :( Sorry about that.

Comment (basvodde) [inline tests/CppUTestExt/MockSupportTest.cpp]: Ref: http://stackoverflow.com/questions/2929543/instance-where-embedded-c-compilers-dont-support-multiple-inheritance

**Comment by arstrube** [comment]
Okay, the answer right at the end clinches it. Even if it is "only" Kernel development. My only option is the two list approach. I'm relying on you to reduce it to one list again :)

**Comment by basvodde** [comment]
Then we'll need a list per class...

**Comment by arstrube** [comment]
That is so. Unless you have a better idea, of course...

On the bright side, it was like that before I refactored it to use two interfaces :-). 

The stupid thing is that a list is needed only for two reasons:
1. Must keep track in order to delete the custom thingies
2. The c functions have a different type to the c++ "convenience" functions.

**Comment by arstrube** [comment]
@basvodde any bright ideas yet? Or a decision on the lesser evil, perhaps?

**Comment by basvodde** [comment]
Done.

**Pull request merged**